proto/p4/p4runtime.proto

```diff
@@ -314,6 +315,33 @@ message CounterData {
   int64 packet_count = 2;
 }
 
+//------------------------------------------------------------------------------
+// Only one instance of a Packet Replication Engine (PRE) is expected in the
+// P4 pipeline. Hence, no instance id is needed to access the PRE.
+message PreEntry {
+  oneof pre_entry {
+    MulticastGroupEntry multicast_group_entry = 1;
+    CloneSessionEntry clone_session_entry = 2;
+  }
+}
+
+// The egress instance must be unique for each replica in a given multicast
+// group entry.
+message MulticastGroupEntry {
+  bytes multicast_group = 1;
+  message Replica {
+    bytes port = 1;
```

Comment:
As discussed in person, I believe the notion of port may be too restrictive for this (as for the action profile member watch attribute). I would keep the `bytes` type but rename it (maybe `destination`?). We should discuss this topic as the next P4 API WG meeting, along with the watch attribute,

Reply:
Yes, let's discuss this further. IMHO, it would be cleaner to draw a boundary and keep everything in the P4Runtime service at the switch scope: hence the notion of singletons and trunks is OK, but no higher-level abstractions like routes. If we want to keep a watch on routes, that would need to be managed by the controller, not the P4Runtime service.

Reply:
Neither P4 nor PSA give special status to LAGs. So it should be either singleton hardware ports only (which do have special status) or a generic mechanism that is able to support LAG among other protocols / abstractions. I'm for consistency.

Reply:
PSA defines the architecture of a reconfigurable switching ASIC. If we were to treat P4Runtime as a runtime API to a PSA-targeted P4 program, then it will turn into a very thin SDK of a reconfigurable switching ASIC. At which point, it will be not be very useful to an SDN controller, that needs an interface to a switch, not ASIC. I am hoping we can settle on the right abstraction and scope of P4Runtime like the PSA group did. Let's discuss this more at the WG meeting. This dichotomy between switch and ASIC is the source of endless confusion.

Reply:
I am going to flip-flop here and propose that the egress port should be a singleton by default for both multicast and clone cases. If we want to do translation/resolution/watch in the p4runtime service, we should do so through annotations.